@xen-orchestra/rest-api/src/users/user.controller.mts

```diff
@@ -0,0 +1,50 @@
+import { Example, Get, Path, Query, Request, Response, Route, Security, Tags } from 'tsoa'
+import { Request as ExRequest } from 'express'
+import { provide } from 'inversify-binding-decorators'
+import { notFoundResp, unauthorizedResp, Unbrand } from '../open-api/common/response.common.mjs'
+import { XoController } from '../abstract-classes/xo-controller.mjs'
+import type { WithHref } from '../helpers/helper.type.mjs'
+import { XoUser } from '@vates/types'
+import { partialUsers, user, userIds } from '../open-api/oa-examples/user.oa-example.mjs'
+
+@Route('users')
+@Security('*')
+@Response(unauthorizedResp.status, unauthorizedResp.description)
+@Tags('users')
+@provide(UserController)
+export class UserController extends XoController<XoUser> {
+  // --- abstract methods
+  getAllCollectionObjects(): Promise<XoUser[]> {
+    return this.restApi.xoApp.getAllUsers()
+  }
+  getCollectionObject(id: XoUser['id']): Promise<XoUser> {
+    return this.restApi.xoApp.getUser(id)
+  }
+
+  /**
+   * @example fields "email,id"
+   * @example filter "permission:admin"
```

Comment:
Add the filtered property in the fields.
Also, we try to not use `email` anymore. Use `name` instead.
Don't forget to update your examples.
```suggestion
   * @example fields "permission,name,id"
   * @example filter "permission:admin"
```